zerver/views/typing.py

```diff
@@ -38,13 +38,27 @@ def send_notification_backend(
         if topic is None:
             raise JsonableError(_("Missing topic"))
 
+        if not user_profile.send_stream_typing_notifications:
+            return json_success(
+                {
+                    "msg": "Sharing stream typing notifications turned off. This request will be ignored."
```

Comment:
Does it make sense to do it this way? The other option of raising an error might break things in other clients.

Also, this string might benefit a bit from rephrasing it to be more concise.

Reply:
Can you ask about this question in the "api design" stream?

Reply:
Changed to raise errors.

(for later reference: link to the discussion: https://chat.zulip.org/#narrow/stream/378-api-design/topic/typing.20notifications.20endpoint